src/qt/guiutil.cpp

```diff
@@ -815,32 +815,33 @@ QString formatDurationStr(int secs)
     return strList.join(" ");
 }
 
+QString serviceFlagToStr(const quint64 mask, const int bit)
+{
+    switch (ServiceFlags(mask)) {
+    case NODE_NONE: abort();  // impossible
+    case NODE_NETWORK:         return "NETWORK";
+    case NODE_GETUTXO:         return "GETUTXO";
+    case NODE_BLOOM:           return "BLOOM";
+    case NODE_WITNESS:         return "WITNESS";
+    case NODE_NETWORK_LIMITED: return "NETWORK_LIMITED";
+    // Not using default, so we get warned when a case is missing
+    }
+    if (bit < 8) {
+        return QString("%1[%2]").arg("UNKNOWN").arg(mask);
+    } else {
+        return QString("%1[2^%2]").arg("UNKNOWN").arg(bit);
```

Comment:
Why do bits `>=8` get `2^` but `<8` don't?

Reply:
To use the same values that were shown previously for such bits. I'm assuming eventually we'll have all 8 assigned to literal strings, and the special case can go away then.

Reply:
(Also because 9223372036854775808 is a pretty big number)

Reply:
OH! I get it now.
But personally I'd show 2^X for all bits then from now on, there's no need for complete version to version consistency in the GUI.